examples/todocontroller/e2e-test.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash -e
+
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+LANGUAGE="$1"
+
+function get_lb_ip() {
+    kubectl --namespace=${USER} get service hello-grpc-staging \
+	-o jsonpath='{.status.loadBalancer.ingress[0].ip}'
+}
+
+function create() {
+    bazel build examples/todocontroller/${LANGUAGE}:staging.create
+    bazel-bin/examples/todocontroller/${LANGUAGE}/staging.create
+    bazel build examples/todocontroller:example-todo.create
+    bazel-bin/examples/todocontroller/example-todo.create
+}
+
+function check_msg() {
+    bazel build examples/todocontroller:example-todo.describe
+
+    OUTPUT=$(./bazel-bin/examples/todocontroller/example-todo.describe)
+    echo Checking response from service: "${OUTPUT}" matches: "DEMO$1<space>"
+    echo "${OUTPUT}" | grep "DEMO$1[ ]"
+}
+
+function edit() {
+    ./examples/todocontroller/${LANGUAGE}/edit.sh "$1"
+}
+
+function update_controller() {
+    bazel build examples/todocontroller/${LANGUAGE}:controller-deployment.replace
+    bazel-bin/examples/todocontroller/${LANGUAGE}/controller-deployment.replace
+}
+
+function update_todo() {
+    bazel build examples/todocontroller:example-todo.replace
+    bazel-bin/examples/todocontroller/example-todo.replace
+}
+
+function delete() {
+    bazel run examples/todocontroller/example-todo.describe
+    bazel run examples/todocontroller/${LANGUAGE}:controller-deployment.describe
+
+    bazel build examples/todocontroller/example-todo.delete
```

Comment:
Why not bazel run everything in here, instead of bazel building then executing?

Reply:
I want coverage of separate build/run working :)